api/src/services/items.ts

```diff
@@ -730,4 +755,22 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 
 		return await this.createOne(data, opts);
 	}
+
+	/**
+	 * Validate primary keys based on its type
+	 */
+	validatePrimaryKeys(primaryKeyField: string, keys: PrimaryKey | PrimaryKey[]): void {
+		if (Array.isArray(keys)) {
+			for (const key of keys) {
+				this.validatePrimaryKeys(primaryKeyField, key);
+			}
+		} else {
+			const primaryKeyFieldType = this.schema.collections[this.collection].fields[primaryKeyField].type;
+			if (primaryKeyFieldType === 'uuid' && !validateUUID(String(keys))) {
+				throw new ForbiddenException();
+			} else if (primaryKeyFieldType === 'integer' && !Number.isInteger(Number(keys))) {
+				throw new ForbiddenException();
+			}
+		}
+	}
```

Comment:
Can we make this a separate util function rather than a method on the items service?

Reply:
Done.